core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -19,14 +23,14 @@ final case class NonEmptyVector[A] private (toVector: Vector[A]) {
 
   /** Updates the element at the index, if it exists */
   def updated(i: Int, a: A): Option[NonEmptyVector[A]] =
-    if (toVector.isDefinedAt(i)) Some(NonEmptyVector(toVector.updated(i, a))) else None
+    if (toVector.isDefinedAt(i)) Some(new NonEmptyVector(toVector.updated(i, a))) else None
```

Comment:
I'm a bit nervous about `isDefinedAt` I'm afraid this could be coming from `Function` and not be specialized well for vector. What about:
```scala
@inline
private def inBounds(i: Int): Boolean = 0 <= i && i <= toVector.size
```
And use it here and on `get`?